cas-server-support-pac4j/src/main/java/org/jasig/cas/support/pac4j/authentication/handler/support/AbstractPac4jAuthenticationHandler.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.jasig.cas.support.pac4j.authentication.handler.support;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jasig.cas.authentication.BasicCredentialMetaData;
+import org.jasig.cas.authentication.DefaultHandlerResult;
+import org.jasig.cas.authentication.PreventedException;
+import org.jasig.cas.authentication.HandlerResult;
+import org.jasig.cas.authentication.handler.support.AbstractPreAndPostProcessingAuthenticationHandler;
+import org.jasig.cas.support.pac4j.authentication.principal.ClientCredential;
+import org.pac4j.core.profile.UserProfile;
+import org.springframework.beans.factory.annotation.Value;
+
+import javax.security.auth.login.FailedLoginException;
+import java.security.GeneralSecurityException;
+
+/**
+ * Abstract pac4j authentication handler which builds the CAS handler result from the pac4j user profile.
+ *
+ * @author Jerome Leleu
+ * @since 4.1.0
+ */
+public abstract class AbstractPac4jAuthenticationHandler extends AbstractPreAndPostProcessingAuthenticationHandler {
+
+    /**
+     * Whether to use the typed identifier (by default) or just the identifier.
+     */
+    @Value("${clientAuthenticationHandler.typedIdUsed:true}")
```

Comment:
Also, I am assuming this is added to the docs, but it should definitely be added to the cas.properties file of the webapp.